src/deploy/deployer_verifier.ts

```diff
@@ -0,0 +1,24 @@
+import { DeployFunction } from "hardhat-deploy/types";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+import { contractNames } from "../ts/deploy";
+
+const deployVerifierContract: DeployFunction = async function (
+  hre: HardhatRuntimeEnvironment,
+) {
+  const { deployments, getNamedAccounts } = hre;
+  const { deployer } = await getNamedAccounts();
+  const { deploy } = deployments;
+
+  const { allowListOffChainManaged } = contractNames;
+
+  await deploy(allowListOffChainManaged, {
+    from: deployer,
+    gasLimit: 8000000,
+    args: [],
+    log: true,
+    deterministicDeployment: true,
```

Comment:
Here deterministic deployment are dangerous!
Two reasons:
- contract is `Ownable`, which means that the deployer is the initial owner. However, the deployer is a contract, not the EOA deployer address.
- it's not a unique contract but it's supposed to be deployed at every auction. Since it has no deployment input, there can only be one that is deployed with this configuration, and a user trying to copy this function would see deployment passing because the already deployed contract was reused. This contract is however owned by someone else!

Reply:
autssch